### DISCUSSION TITLE
1.0.0rc1

### DISCUSSION THREAD

**Discussion by OStrama** [OStrama/weishaupt_modbus#25] [Announcements]
The 1st release candidate for the new version 1.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/OStrama/weishaupt_modbus/releases/tag/1.0.0rc1'>1.0.0rc1</a>.</em>

**Comment by exp1338** [comment]
The way to upgrade from 0.0.7 is still the same like disscripted for an upgrade to 0.0.8?

**Comment by OStrama** [comment]
When you upgrade from 0.0.7 yes. From 0.0.8 no further action is required. Although there are some new parameters, the integration will behave than before, since the parameters are set to defaults. However, if you want to use new features like name prefixes, you‘d need to re-install the integration. But please be aware, that new entity names will break the long-term statistics.

**Reply by exp1338** [reply]
Thanks works fine. No need to adjust dashboard after upgrade. Hopefully next release will be pop up as an update inside HACS.

**Reply by OStrama** [reply]
When we are ready with testing, there will be an official release that should pop-up in HACS

**Comment by MadOne** [comment]
What to test:

- [x] Update from all "OStrama" versions to 1.
- [x] Update from all MadOne versions to 1.</del>
- [x] Setup clean new entry. 
- [x] Test and compare all values.
- [x] Test modification of editable values. 
- [x] Test config options. 
- [x] Test with non english nor german version. 

Edit: please feel free to suggest tests

**Reply by MadOne** [reply]
Need help with this as i can only read english and german. Can anyone test the installation with another language?

**Reply by MadOne** [reply]
Heizkreis Betriebsart: 
we:
![image](https://github.com/user-attachments/assets/13b3e379-9bbb-4692-afc3-0e399d461bf4)
WEM Portal
![image](https://github.com/user-attachments/assets/ba5311a7-5ba2-4501-a8e0-e675414e5ee5)


Betriebsstunden E1: We:0 WEM: 5:00:00

Wärmepumpe Anforderung:
![image](https://github.com/user-attachments/assets/00f1521e-6557-4c26-bef3-111da6eb30a8)

Sensors without values are displayed with 0 instead of beeing filtered out.

**Reply by OStrama** [reply]
Rebuil case 32768 --> None

**Reply by MadOne** [reply]
> Rebuil case 32768 --> None

I'm sorry, but i don't unterstand what you mean.

**Reply by OStrama** [reply]
A, sorry, I switched back to treating a raw temp value of 32768 as not valid..

**Reply by MadOne** [reply]
Ah i see, this was introduced at the below zero sensor behavour. Is the 0°C correctly displayed with your fix?

**Reply by OStrama** [reply]
I‘m not sure, have to wait for 0°..

**Reply by MadOne** [reply]
What about logging raw modbus sendor data?

**Reply by OStrama** [reply]
yes, that's what I meant. Looking at the recorder data. If we see "1 0.5 0.5 0 -0.5 -1 " everything is OK. If " 1 0.5 0.5 Unbekannt -0.5 -1" then 32768 == Zero. But I assume that 0 == 0 and 32768 == -0 (https://de.wikipedia.org/wiki/Einerkomplement)
So we could treat -0 as not available..

**Reply by OStrama** [reply]
switched to Italiano, our integration falls back to the content of strings.json. I think this is the expected behavior, if a language is not available..

**Reply by MadOne** [reply]
Perfect. You ticked all the checkboxes? So its time for 1.0.0 !

**Reply by OStrama** [reply]
Yes, I‘ll create it today in the break ;-)

**Comment by OStrama** [comment]
Did you manage to install HACS on the Devcontainer? I always get stuck here:

![image](https://github.com/user-attachments/assets/1cb5125d-532f-4444-95c4-e54346ae8f1e)

**Reply by MadOne** [reply]
Didnt try yet. Can do if you want me to

**Reply by MadOne** [reply]
Worked for me:
https://hacs.xyz/docs/use/download/download/#to-download-hacs-core
I installed it with the install script. (Core).

**Reply by OStrama** [reply]
OK, thanks! Unfortunately not on my side, continue trying..

**Comment by MadOne** [comment]
For users of OStrama's weishaupt_wbb integration:
Uninstall existing "weishaupt_wbb" installation, answer "integration and all entities of it will be deleted" with"yes"
Restart home assistant
Install new weishaupt_wbb integration
You will get a new integration with the same name
the sensor entities will be the same than before

Is this still up to date?

**Reply by OStrama** [reply]
For versions before 0.0.8 yes.

**Reply by MadOne** [reply]
What happens if you just upgrade?

**Reply by OStrama** [reply]
You‘ll get two separate integrations, because of two different device prefixes I guess. But maybe I have to test it..

**Reply by OStrama** [reply]
tested it. You get a new integration and the old one stops with an error message "Integration not available"

**Reply by MadOne** [reply]
Perfect. So we wont have any Problem if they do it "wrong"

**Comment by MadOne** [comment]
I updated the migration from "MadOne" version.

**Comment by OStrama** [comment]
1.0.4 is out, so I close this thread
